lib/src/handle_access_token_response.dart

```diff
@@ -24,45 +22,38 @@ const _expirationGrace = const Duration(seconds: 10);
 /// OAuth2 flow.
 /// 
 /// The scope strings will be separated by the provided [delimiter].
+///
+/// [getParameters] may be a function used to parse parameters out of responses from hosts that do not
+/// respond with application/json or application/x-www-form-urlencoded bodies. Use this
+/// to support authorization servers that do not completely follow the OAuth 2.0 standard.
 Credentials handleAccessTokenResponse(
     http.Response response,
     Uri tokenEndpoint,
     DateTime startTime,
     List<String> scopes,
-    String delimiter) {
-  if (response.statusCode != 200) _handleErrorResponse(response, tokenEndpoint);
+    String delimiter,
+    {Map<String, dynamic> getParameters(String contentType, String body)}) {
+  getParameters ??= parseJsonParameters;
 
-  validate(condition, message) =>
-      _validate(response, tokenEndpoint, condition, message);
+  if (response.statusCode != 200) _handleErrorResponse(response, tokenEndpoint, getParameters);
 
   var contentTypeString = response.headers['content-type'];
-  var contentType = contentTypeString == null
-      ? null
-      : new MediaType.parse(contentTypeString);
-
-  // The spec requires a content-type of application/json, but some endpoints
-  // (e.g. Dropbox) serve it as text/javascript instead.
-  validate(contentType != null &&
-      (contentType.mimeType == "application/json" ||
-       contentType.mimeType == "text/javascript"),
-      'content-type was "$contentType", expected "application/json"');
 
   Map<String, dynamic> parameters;
+
   try {
-    var untypedParameters = JSON.decode(response.body);
-    validate(untypedParameters is Map,
-        'parameters must be a map, was "$parameters"');
-    parameters = DelegatingMap.typed(untypedParameters);
-  } on FormatException {
-    validate(false, 'invalid JSON');
+    parameters = getParameters(contentTypeString, response.body);
```

Comment:
It seems wrong that we're passing the full content type here, but only the mime type in `_handleErrorResponse()` below. Why not just pass in the parsed `MediaType` in both cases?

Also, if whatever we pass in can be `null`, that should be clearly documented.